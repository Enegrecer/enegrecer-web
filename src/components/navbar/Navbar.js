import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import M from 'materialize-css';
import logo from '../../imagens/logo.svg';
import './navbar.css';

class Navbar extends Component {
  componentDidMount() {
    const elem = document.querySelector('.sidenav');
    M.Sidenav.init(elem, {
      edge: 'left',
      inDuration: 250
    });
  }

  navigationLinks() {
    const links = [
      { id: 1, to: '/sobre', label: 'SOBRE' },
      { id: 2, to: '/depois-denuncia', label: 'E DEPOIS DA DENÚNCIA?' },
      { id: 3, to: '/coletivo-enegrecer', label: 'O COLETIVO ENEGRECER' },
      {
        id: 4,
        to: '/denunciar',
        label: 'DENUNCIAR',
        className: 'denunciar'
      }
    ];

    return (
      <ul>
        {links.map(link => (
          <li key={link.id}>
            <Link
              className={link.className}
              to={link.to}
              onClick={this.escondeMenu}
              onKeyDown={this.escondeMenu}
            >
              {link.label}
            </Link>
          </li>
        ))}
      </ul>
    );
  }

  render() {
    return (
      <div className="navbar">
        <div className="navbar-fixed">
          <nav>
            <div className="nav-wrapper">
              <Link href="/" to="/" className="logo-desktop brand-logo">
                <img src={logo} alt="Logo Verdade Seja Dita" />
              </Link>
              <a href="#!" data-target="slide-out" className="sidenav-trigger">
                <i className="material-icons">menu</i>
              </a>
              <ul className="right hide-on-med-and-down">
                <div key={7} className="links">
                  {this.navigationLinks()}
                </div>
              </ul>
            </div>
          </nav>
        </div>

        <ul id="slide-out" className="sidenav">
          <div className="logo-wrapper">
            <div className="logo-mobile">
              <Link href="/" to="/">
                <img src={logo} alt="Logo Verdade Seja Dita" />
              </Link>
            </div>
          </div>
          <li>
            <a className="waves-effect botao-denunciar" href="/denunciar">
              <i className="material-icons">add_comment</i>Denunciar
            </a>
          </li>
          <li>
            <a className="waves-effect" href="/depois-denuncia">
              <i className="material-icons">help</i>E depois da denúncia?
            </a>
          </li>
          <li>
            <a className="waves-effect" href="coletivo-enegrecer">
              <i className="material-icons">supervisor_account</i>Coletivo Enegrecer
            </a>
          </li>
          <li>
            <a className="waves-effect" href="/sobre">
              <i className="material-icons">info</i>Sobre
            </a>
          </li>
        </ul>
      </div>
    );
  }
}

export default Navbar;
