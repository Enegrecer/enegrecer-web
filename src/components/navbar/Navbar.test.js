import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('deve ter a mesma estrutura de UI', () => {
    const navbar = shallow(<Navbar />);
    const navbarJson = toJson(navbar);

    expect(navbarJson).toMatchSnapshot();
  });
});
