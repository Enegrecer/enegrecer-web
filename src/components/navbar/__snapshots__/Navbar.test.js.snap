// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar deve ter a mesma estrutura de UI 1`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav>
      <div
        className="nav-wrapper"
      >
        <Link
          className="logo-desktop brand-logo"
          href="/"
          replace={false}
          to="/"
        >
          <img
            alt="Logo Verdade Seja Dita"
            src="logo.svg"
          />
        </Link>
        <a
          className="sidenav-trigger"
          data-target="slide-out"
          href="#!"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
        <ul
          className="right hide-on-med-and-down"
        >
          <div
            className="links"
            key="7"
          >
            <ul>
              <li
                key="1"
              >
                <Link
                  replace={false}
                  to="/sobre"
                >
                  SOBRE
                </Link>
              </li>
              <li
                key="2"
              >
                <Link
                  replace={false}
                  to="/depois-denuncia"
                >
                  E DEPOIS DA DENÚNCIA?
                </Link>
              </li>
              <li
                key="3"
              >
                <Link
                  replace={false}
                  to="/coletivo-enegrecer"
                >
                  O COLETIVO ENEGRECER
                </Link>
              </li>
              <li
                key="4"
              >
                <Link
                  className="denunciar"
                  replace={false}
                  to="/denunciar"
                >
                  DENUNCIAR
                </Link>
              </li>
            </ul>
          </div>
        </ul>
      </div>
    </nav>
  </div>
  <ul
    className="sidenav"
    id="slide-out"
  >
    <div
      className="logo-wrapper"
    >
      <div
        className="logo-mobile"
      >
        <Link
          href="/"
          replace={false}
          to="/"
        >
          <img
            alt="Logo Verdade Seja Dita"
            src="logo.svg"
          />
        </Link>
      </div>
    </div>
    <li>
      <a
        className="waves-effect botao-denunciar"
        href="/denunciar"
      >
        <i
          className="material-icons"
        >
          add_comment
        </i>
        Denunciar
      </a>
    </li>
    <li>
      <a
        className="waves-effect"
        href="/depois-denuncia"
      >
        <i
          className="material-icons"
        >
          help
        </i>
        E depois da denúncia?
      </a>
    </li>
    <li>
      <a
        className="waves-effect"
        href="coletivo-enegrecer"
      >
        <i
          className="material-icons"
        >
          supervisor_account
        </i>
        Coletivo Enegrecer
      </a>
    </li>
    <li>
      <a
        className="waves-effect"
        href="/sobre"
      >
        <i
          className="material-icons"
        >
          info
        </i>
        Sobre
      </a>
    </li>
  </ul>
</div>
`;
